README.md

```diff
@@ -70,3 +70,96 @@ const parseServerOptions = {
   /* Other Parse Server options */
 }
 ```
+
+### Apple configuration
+
+Delivering push notifications to Apple devices can be done either via Apple APNS (Apple Push Notification Service), or via FCM (Firebase Cloud Messaging) service. Note that each category of Apple devices require their own configuration section. Parse Server Push Adapter currently supports these types of Apple devices:
+
+- `ios` -> iPhone, iPad, and iPod touch apps
+- `osx` -> macOS, and macCatalyst apps
+- `tvos` -> tvOS apps
+
+
+Apple Push Notification Service requires a p8 encoded private key that can be generated and downloaded in the Apple Developer Center under Certificates > Identifiers & Profiles.
+
+Google Firebase Cloud Messaging requires a Firebase Service Account JSON encoded private key that can be downloaded in the Firebase Console under your project's Settings > Cloud Messaging tab.
+
+Here is an example configuration for both methods:
+
+```js
+const PushAdapter = require('@parse/push-adapter').default;
+const parseServerOptions = {
+  push: {
+    adapter: new PushAdapter({
+      ios: {
+        /* Deliver push notifications to iOS devices via Apple APNS */
+        /* You will need app id, team id, and auth key available on developer.apple.com/account */
+        token: {
+          key: __dirname + '/AuthKey_XXXXXXXXXX.p8',
+          keyId: "XXXXXXXXXX",
+          teamId: "AAAAAAAAAA"
+        },
+        topic: 'com.example.yourawesomeapp',
+        production: true
+      },
+      osx: {
+        /* Deliver push notifications to macOS devices via Google FCM */
+        /* You will need admin key available on console.firebase.google.com */
+        firebaseServiceAccount: __dirname + "/your-awesome-app-firebase-adminsdk-abcd-efgh.json"
+      }
+    })
+  },
+  /* Other Parse Server options */
+}
+```
+
+### Android configuration
+
+Delivering push notifications to Android devices can be done via FCM (Firebase Cloud Messaging) service.
+
+Google Firebase Cloud Messaging requires Firebase Service Account json encoded private key that can be downloaded in Firebase Console under your project Settings, and Cloud Messaging tab.
+
+Here is an example configuration:
+
+```js
+const PushAdapter = require('@parse/push-adapter').default;
+const parseServerOptions = {
+  push: {
+    adapter: new PushAdapter({
+      android: {
+        /* Deliver push notifications to Android devices via Google FCM */
+        /* You will need admin key available on console.firebase.google.com */
+        firebaseServiceAccount: __dirname + "/your-awesome-app-firebase-adminsdk-abcd-efgh.json"
+      }
+    })
+  },
+  /* Other Parse Server options */
+}
+```
+
+#### Migration from GCM to FCM (June 2024)
+
+Sending push notifications to Android devices was originally implemented using GCM (Google Cloud Messaging) API. GCM API was deprecated on June 20 2023 and will stop working after June 20 2024.
```

Comment:
Is that correct? GCM has been decommissioned a while ago. I think we are migrating from FCM legacy API to the HTTP v1 API, which btw is also already deprecated, but I don't benefit a sunset date has been announced yet.

See https://firebase.google.com/docs/cloud-messaging/migrate-v1

We could actually add that link to the docs.

Reply:
Oh you may actually be right, I am honestly completely confused by the Google naming and was not involved in the Android push until recently... feel free to change as you understand it, I do not want to spread false info...